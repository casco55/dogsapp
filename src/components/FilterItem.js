import React from 'react';
import { useDispatch } from 'react-redux';
import { removeFilterItem } from '../actions/selectores';


export const FilterItem = ({ item }) => {
    const dispatch = useDispatch();
    const handleRemoveItem = () => {
        dispatch(removeFilterItem(item))
    }
    return (
        <>
            <div className="d-flex justify-content-between col-7 col-md-2 my-2 mx-2 px-2 py-1 my-auto border border-dark rounded-pill borderFIlterItem texto"   >
                {item}
                <i className="mr-1 far fa-times-circle my-auto" onClick={ handleRemoveItem }></i>
            </div>
        </>
    )
}
