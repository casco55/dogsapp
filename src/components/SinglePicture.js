import React from 'react';
import PropTypes from 'prop-types';

export const SinglePicture = ({item, selectedPicture}) => {
    return (
        <>
            <div className="container-fluid d-flex flex-column">
                <h3 className="text-center d-md-none mt-1 mt-md-2">{item}</h3>
                <h2 className="text-center d-none d-md-block mt-1 mt-md-2">{item}</h2>
                <img src={selectedPicture} alt="Imagen de raza" className="img-fluid h-75 mx-auto my-auto" />
            </div>
        </>
    )
}

SinglePicture.propTypes = {
    item: PropTypes.string.isRequired,
    selectedPicture: PropTypes.string.isRequired,
}
