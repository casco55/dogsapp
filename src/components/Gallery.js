import React, { useState } from 'react'
import { SinglePicture } from './SinglePicture';

export const Gallery = ({item, images}) => {

    const initialState = {
        selectedPicture: '',
    }   

    const [values, setValues] = useState(initialState);
    const { selectedPicture } = values;
    console.log(selectedPicture);

    const scrollLeft = () => {
        const container = document.getElementById(item);
                let scrollAmount = 0;
                var slideTimer = setInterval(function(){
                    container.scrollLeft -= 100;
                    scrollAmount += 10;
                    if(scrollAmount >= 80){
                        window.clearInterval(slideTimer);
                    }
                }, 50);
        
    }
    const scorllRight = () => {
        const container = document.getElementById(item);
                let scrollAmount = 0;
                var slideTimer = setInterval(function(){
                    container.scrollLeft += 100;
                    scrollAmount += 10;
                    if(scrollAmount >= 80){
                        window.clearInterval(slideTimer);
                    }
                }, 50);
    }



    return (
        <>
            <h5 className="mt-3 ml-2">{item}:</h5>
            <div className="d-flex col-12">
                <i className="d-none d-md-block fas fa-2x fa-angle-left my-auto" onClick={ scrollLeft }></i>
                <div className="d-flex flex-row overflow-y my-2 mx-md-2" id={item}>
                    {images.length > 0 && 

                        images.map((image) => (
                            <div className="col-4 col-md-2 col-lg-2 my-auto" key={image}>
                                <img src={image} alt="Imagen de perro" className="img-fluid" onClick={() => setValues({...values, selectedPicture: image})}/>
                            </div>
                        ))
                        
                    }
                </div>
                <i className="d-none d-md-block fas fa-2x fa-angle-right my-auto" onClick={ scorllRight }></i>
            </div>

            {selectedPicture !== '' &&
                <div className="fixed-top bg-light col-lg-12 pb-5 d-flex flex-column h-100 over">
                    <div className="mr-0 ml-auto">
                            <i className="far fa-times-circle fa-3x" onClick={ () => setValues(initialState) }></i>
                        </div>        
                    <SinglePicture selectedPicture ={selectedPicture} item={item} />
                     
                </div>
            }
        </>
    )
}
