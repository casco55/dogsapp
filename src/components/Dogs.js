import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getAllBreeds } from '../actions/selectores';
import { Header } from '../ui/Header';
import { FilterItem } from './FilterItem';
import { Gallery } from './Gallery';


export const Dogs = () => {

    const dispatch = useDispatch();
    useEffect(() => {
        dispatch( getAllBreeds() );
    }, [dispatch]);
    const { optionsList } = useSelector(state => state.list);
    const { searchList } = useSelector(state => state.list);
    
    return (
        <>
            <div className="sticky-top">
                {optionsList !== undefined && 
                    <Header optionsList={optionsList} />
                } 
                {searchList.length > 0 && 
                    <div className="d-flex flex-row sticky-top overflow-y bg-light">
                        {searchList.map((item) => (
                            <FilterItem key={item.breed} item={item.breed} />
                        ))}
                    </div>
                }
            </div>
            {searchList.length > 0 && 
                <div className="d-flex flex-column mt-3 overflow-auto ">
                    {searchList.map((item) => (
                        <Gallery key={item.breed} item={item.breed} images={item.images} />
                    ))}
                </div>
            }
            
        </>
    )
}
