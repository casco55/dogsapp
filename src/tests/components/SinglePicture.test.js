import { shallow } from 'enzyme';
import React from 'react';


import { SinglePicture } from '../../components/SinglePicture';

describe('Pruebas en <SinglePicture />', () => {

        const item = 'raza';
        const selectedPicture = 'https://example.com';        
        const wrapper = shallow( <SinglePicture item={item} selectedPicture={selectedPicture} /> );

    test('debe mostrar el componente correctamente ', () => {

        
        expect( wrapper ).toMatchSnapshot();

    })

    test('debe tener un h3 y un h2 con el item ', () => {
        const h3 = wrapper.find('h3');
        const h2 = wrapper.find('h2');
        expect( h3.text().trim() && h2.text().trim() ).toBe( item );
    })
    
    test('Debe tener una imagen con selectedPicture', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( selectedPicture );
    })
    
})
