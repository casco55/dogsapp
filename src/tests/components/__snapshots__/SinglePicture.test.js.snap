// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SinglePicture /> debe mostrar el componente correctamente  1`] = `ShallowWrapper {}`;
